### PR TITLE
WIP: source: csv: Enable setting repo src_url 

### PR DESCRIPTION
**Pull request by sudharsana-kjl**
A fix for issue #48

### CODE REVIEW

File: dffml/source/csv.py [hunk 1]
```diff
@@ -59,9 +59,14 @@ class CSVSource(FileSource, MemorySource):
                         }
                     }
                 )
-            # Create the repo with the source URL being the row index
-            repo = Repo(str(i), data=repo_data)
-            i += 1
+            if self.config.src_url == key:
```
Comment (pdxjohnny) [inline dffml/source/csv.py]: We'll change the default for `config.src_url` to `None`, and then check if it was set by checking if it's not None

```python
if self.config.src_url is not None and self.config.src_url == key:
```

File: dffml/source/file.py [hunk 2]
```diff
@@ -114,11 +115,16 @@ def args(cls, args, *above) -> Dict[str, Arg]:
             "readonly",
             Arg(type=bool, action="store_true", default=False),
         )
+        cls.config_set(args,
+                       above,
+                       "src_url",
+                       Arg(type=str, default="DefaultSourceURL"))
```
Comment (pdxjohnny) [inline dffml/source/file.py]:
```python
Arg(type=str, default=None)
```

File: dffml/source/file.py [hunk 3]
```diff
@@ -114,11 +115,16 @@ def args(cls, args, *above) -> Dict[str, Arg]:
             "readonly",
             Arg(type=bool, action="store_true", default=False),
         )
+        cls.config_set(args,
+                       above,
+                       "src_url",
```
Comment (pdxjohnny) [inline dffml/source/file.py]: Let's name this `key` instead of `src_url`

File: dffml/source/csv.py (hunk 1 above)
Comment (sudharsana-kjl) [reply dffml/source/csv.py]: Done :+1:

File: dffml/source/file.py (hunk 2 above)
Comment (sudharsana-kjl) [reply dffml/source/file.py]: okay :+1:

File: dffml/source/file.py (hunk 3 above)
Comment (sudharsana-kjl) [reply dffml/source/file.py]: okay :+1:

File: CHANGELOG.md [hunk 4]
```diff
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   config for them.
 - shouldi example uses updated `MemoryOrchestrator.basic_config` method and
   includes more explanation in comments.
+- enable setting repo src_url
```
Comment (pdxjohnny) [inline CHANGELOG.md]: Please capitalize enable

```markdown
- CSVSource allows for setting the Repo's `src_url` from a csv column
```

File: tests/source/test_csv.py [hunk 5]
```diff
@@ -9,3 +9,10 @@
 class TestCSVSource(SourceTest, AsyncTestCase):
     async def setUpSource(self, fileobj):
         return CSVSource(FileSourceConfig(filename=fileobj.name))
+
+class CSVTest(SourceTest, AsyncTestCase):
+    async def setUpSource(self, fileobj):
+        return CSVSource(FileSourceConfig(filename=fileobj.name, key="UniqueFeature"))
+
+    async def test_key(self):
+        file =
```
Comment (pdxjohnny) [inline tests/source/test_csv.py]: Now make a csv file, you could just do

```python
with tempfile.NamedTemporaryFile() as fileobj:
    fileobj.write(b"KeyHeader,ValueColumn\n")
    fileobj.write(b"a,42\n")
    fileobj.write(b"b,420\n")
    fileobj.seek(0)
    async with CSVSource(FileSourceConfig(filename=fileobj.name, key="KeyHeader")) as source:
        async with source() as sctx:
        # Check that the values for ValueColumn are correct for each repo (a, b)
```

Comment (sudharsana-kjl) [reply tests/source/test_csv.py]: Oh okay. I've done in a different way. I didnt see this. I'll change it now.

File: tests/source/test_csv.py [hunk 6]
```diff
@@ -15,39 +15,19 @@ class TestCSVSource(SourceTest, AsyncTestCase):
...
+            fileobj.write(b"b,420\n")
+            fileobj.seek(0)
+            async with CSVSource(
+                FileSourceConfig(filename=fileobj.name, key="KeyHeader")
+            ) as source:
+                async with source() as sctx:
+                    repo_a = await sctx.repo("a")
+                    repo_b = await sctx.repo("b")
+                    self.assertEqual(repo_a.data.features["ValueColumn"], 42)
+                    self.assertEqual(repo_b.data.features["ValueColumn"], 420)
```
Comment (pdxjohnny) [inline tests/source/test_csv.py]: Very nice :)

File: tests/source/test_csv.py (hunk 5 above)
Comment (sudharsana-kjl) [reply tests/source/test_csv.py]: Done :+1:

**Pull request merged**